### PR TITLE
Gymnasium/fix sporadic CI failure

### PR DESCRIPTION
**Pull request by dlpbc**
## Summary

- Make Gymnasium position-info tests wait for tracked aircraft with a bounded step loop.
- Fail explicitly if no aircraft are tracked within the step budget.
- Add a Springfield decentralized warm-up regression test for the initial empty-agent state.
- Fix a `route_direct_to` assertion that was accidentally asserting an always-truthy tuple.
- Add a route-direct regression test proving previous fixes are rejected.

### CODE REVIEW

File: bluebird-gymnasium/tests/envs/test_infinite.py [hunk 1]
```diff
@@ -182,25 +182,27 @@ def test_pos_information(view_type: ViewType, env_cls: EnvCls):
...
+
+        # in artificial airspace, we can expect that the first aircraft
+        # at start of the scenario is yet to enter the airspace/sector.
+        assert ret.position_status == PositionStatus.BEFORE_ENTRY
+        assert ret.incomm_status is False
+        assert ret.outcomm_status is False
+        assert ret.dist_to_sector_entry > 0.0
+        assert ret.dist_away_from_sector_exit == 0.0
+        assert ret.dist_away_from_incorrect_sector_exit == 0.0
+        assert ret.incorrect_exit_position is None
```
Comment (AlvaroSierra) [inline bluebird-gymnasium/tests/envs/test_infinite.py]: Should this if statement be within the if statement in line 189, as both seem to do the same check at the same point in time.

File: bluebird-gymnasium/tests/envs/test_infinite.py [hunk 2]
```diff
@@ -182,25 +182,27 @@ def test_pos_information(view_type: ViewType, env_cls: EnvCls):
...
-    assert ret.incomm_status is False
-    assert ret.outcomm_status is False
-    assert ret.dist_to_sector_entry > 0.0
-    assert ret.dist_away_from_sector_exit == 0.0
-    assert ret.dist_away_from_incorrect_sector_exit == 0.0
-    assert ret.incorrect_exit_position is None
+    # forward the simulation to the time when at least one aircraft is being
+    # tracked
+    done = False
+    while not done:
```
Comment (AlvaroSierra) [inline bluebird-gymnasium/tests/envs/test_infinite.py]: Could we turn this from a while loop, to a for loop with a runtime limit, to ensure it would not run forever in a runner without ever reporting an error.

If this loop never triggers the break it should fail the test as well, therefore the break would be a return statement.

**Review by AlvaroSierra — CHANGES_REQUESTED** [review]
Thank you Ese, a few comments to address which apply to all the changed tests.

File: bluebird-gymnasium/tests/envs/test_infinite.py (hunk 2 above)
Comment (dlpbc) [reply bluebird-gymnasium/tests/envs/test_infinite.py]: The `done` flag does exactly that, ensuring it won't run forever. Each episode runs for a finite number of steps. Once the maximum number of steps is reached, then `done` is set to True. However, I'll convert it to a for loop with a fixed budget on the number of steps.

Regarding failing the tests when the break is never triggered, I'll include it. But this is an indicator that something is broken at the sim level rather than at the higher-level gymnasium wrapper.

**Review by nbarlowATI — APPROVED** [review]
Looks good to me!   If/when we merge the sector-agnostic-scenario-manager branch in, where the aircraft are incommed at the boundary fix rather than first fix on route, we should re-check that 100 steps are enough for the aircraft to start being tracked.

**Review by AlvaroSierra — APPROVED** [review]
Looks good to me.

I don't agree with Nick though, if it takes more than 600 seconds for an aircraft to be tracked with the new scenario manager, it is a bug in the scenario, not a problem with the tests.
